Jenkinsfile

```diff
@@ -8,15 +8,13 @@ pipeline {
   parameters {
     choice(name: 'TARGET_PLATFORM', choices: ['r202203', 'r202206', 'r202209', 'r202212', 'r202303', 'latest'], description: 'Which Target Platform should be used?')
     // see https://wiki.eclipse.org/Jenkins#JDK
-    choice(name: 'JDK_VERSION', description: 'Which JDK should be used?', choices: [
-       'temurin-jdk11-latest', 'temurin-jdk17-latest'
-    ])
+    choice(name: 'JDK_VERSION', choices: [ '11', '17' ], description: 'Which JDK version should be used?')
```

Comment:
just glancing at the pr its unclear how this maps to the toolchains

Reply:
so we use toolchains only for 11?

Reply:
tested locally. seems to work when both env vars are set

Reply:
What do you mean by tested locally? This PR does not change the environment variables (those were already changed in the https://github.com/eclipse/xtext/pull/2223 so they were already working). This PR changes the job's parameters names for JDK. Unfortunately, the Jenkins PR does not check that because for security reasons, it still takes the Jenkinsfile from the internal repository not from the external one. We should replay the PR by manually copying the Jenkinsfile. In this case, I think we can avoid that and merge this. Then, I think the first time we'll have to make sure that the jobs for the `main` branch are aware of the changed parameters. We saw in the past, that Jenkins does not react well when you change parameters' names or values' orders ;)

Reply:
this is why i did
https://ci.eclipse.org/xtext/job/xtext/job/HannesWell-jenkinsParams/

Reply:
OK! So you tested the parameters, that's great!
You only mentioned the environment variables, which are independent of the parameters (they are always both set anyway and always), that's why I wanted to be sure.

Reply:
@HannesWell can you confirm this?

Reply:
```
06:52:09  [INFO] --- tycho-surefire-plugin:3.0.4:test (default-test) @ org.eclipse.xtext.purexbase.tests ---
06:52:09  [INFO] Could not find a java toolchain of type jdk, using java from JAVA_HOME instead (/opt/tools/java/temurin/jdk-17/latest/bin/java)
06:52:09  [INFO] Executing Test Runtime with timeout 0, logs (if any) will be placed at: /home/jenkins/agent/workspace/xtext_HannesWell-jenkinsParams/org.eclipse.xtext.purexbase.tests/target/work/data/.metadata/.log
```

Reply:
@cdietrich I once again confirm this.

As I said, that was already tested in the https://github.com/eclipse/xtext/pull/2223 .
This PR only changes the parameter names, it does not affect the toolchain strategy that has already been implemented in https://github.com/eclipse/xtext/pull/2223.

Well, initially it did, but then @HannesWell reverted that part.